### DISCUSSION TITLE
Want to use OpenId Connect authentication in ASP .Net server using Keycloak

### DISCUSSION THREAD

**Discussion by Franco8888** [keycloak/keycloak#19474] [Q&A]
- Want to use Direct Grant Flow
- Want to authenticate a user and get access Token and use the access token to authenticate as Bearer token when calling endpoint on ASP .Net server

Using .Net 7

**Server code in program.cs:**
```
(services.AddAuthentication(options =>
            {
                options.DefaultAuthenticateScheme = CookieAuthenticationDefaults.AuthenticationScheme;
                options.DefaultChallengeScheme = OpenIdConnectDefaults.AuthenticationScheme;
            })
            .AddOpenIdConnect(options =>
            {
                options.Authority = "http://localhost:8080/realms/GT/";
                options.ClientId = "gt_client";
                options.ClientSecret = "******************";
                options.ResponseType = OpenIdConnectResponseType.Code;
                options.GetClaimsFromUserInfoEndpoint = true; // MAYBE FOR ROLES
                options.Scope.Add("openid");
                options.Scope.Add("profile");
                options.RequireHttpsMetadata = false;  // allow HTTP, only in development
                options.SaveTokens = true;
            });)
```

I get the the access token using:
![image](https://user-images.githubusercontent.com/29451172/229136877-557a053e-7d19-46d0-867b-8f9554b0f11c.png)

I am using direct grant flow so the following settings of my Keycloak client are as follows:

- Client authentication: On
- Standard Flow: Off (If I set this setting to on I get the browser requesting login which I do not want, If I set)
- DIrect access grants: On
- Service accounts roles: On
- Rest of the settings are off

**When calling the endpoint on my server I get the following:**
If Standard Flow is OFF:
I get this error: _Client is not allowed to initiate browser login with given response_type. Standard flow is disabled for the client_.
If Standard Flow is ON:
I get the Keycloak browser login which I do not want. I want to use the access_token as bearer token to authenticate.

What would I change to get the desired results?

**Comment by darius-m** [comment]
If I remember correctly, the resource owner password credentials grant (direct access grant) will simply send the ID tokens back to the application; the `code` response type is used for the standard flow, and you should use the `password` grant, as mentioned in the [securing applications](https://www.keycloak.org/docs/latest/securing_apps/index.html#_resource_owner_password_credentials_flow)  guide.
